### ISSUE TITLE
BUG: modified status and diagnostic does not change for buffers that opened by neo-tree

### ISSUE THREAD

**Issue by mrbeardad** [nvim-neo-tree/neo-tree.nvim#1264]
### Did you check docs and existing issues?

- [X] I have read all the docs.
- [X] I have searched the existing issues.
- [X] I have searched the existing discussions.

### Neovim Version (nvim -v)

NVIM v0.9.4

### Operating System / Version

Windows 11

### Describe the Bug

File modified status and diagnostic status does not change for the most file, only works for the file I opened with `nvim works_file`

### Screenshots, Traceback

https://github.com/nvim-neo-tree/neo-tree.nvim/assets/54128430/56721b41-3ad8-4b23-a04a-366519df9623

### Steps to Reproduce

1. `nvim README.md`
2. `:Neotree`
3. modify README.md, modified status show in neotree
4. modify CONTRIBUTING.md, modified status does not show
5. diagnostic status is similar to modified status

### Expected Behavior

modified status and diagnostic status should change for step 4 above.

### Your Configuration

```Lua
if vim.g.vscode then
  return
end

-- bootstrap plugins manager
local plugins_path = vim.fn.stdpath("data") .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(plugins_path) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    "--branch=stable",
    plugins_path,
  })
end
vim.opt.rtp:prepend(plugins_path)

-- load user configs before load plugins
--vim.cmd("runtime! lua/user/configs/*.lua")

require("lazy").setup({
  -- load plugin specifics in directories
  spec = {
    --{ import = "user/plugins" },
    --{ import = "user/langs" },
    { import = "user/debug" },
  },
  defaults = {
    -- do not lazy load plugins by default
    lazy = false,
    -- always use the latest git commit, set to "*" for latest stable version
    version = false,
    -- disable unnecessary plugins for vscode
    cond = function()
      return true
    end,
  },
  install = {
    -- automatically install missing plugins
    missing = true,
    -- try to load one of these colorschemes when starting an installation during startup
    colorscheme = { "tokyonight", "habamax" },
  },
  -- do not automatically check for plugin updates
  checker = { enabled = false },
  -- do not automatically check for config file changes and reload the ui
  change_detection = { enabled = false },
  performance = {
    rtp = {
      disabled_plugins = {
        "gzip",
        -- "matchit",
        -- "matchparen",
        "netrwPlugin",
        "tarPlugin",
        "tohtml",
        "tutor",
        "zipPlugin",
      },
    },
  },
  ui = {
    custom_keys = {
      ["o"] = {
        function(plugin)
          vim.cmd("normal q")
          vim.cmd("Neotree dir=" .. plugin.dir)
        end,
        desc = "Open Plugin Dir",
      },
    },
  },
})



-- debug/neotree.lua
return {
  {
    "nvim-neo-tree/neo-tree.nvim",
    dependencies = {
      "nvim-lua/plenary.nvim",
      "nvim-tree/nvim-web-devicons",
      "MunifTanjim/nui.nvim",
    },
    event = "User DirOpened",
    cmd = "Neotree",
    keys = {
      { "<Leader>e", "<Cmd>Neotree<CR>", desc = "Explorer" },
    },
    opts = {},
  },
}
```

**Comment by mrbeardad** [comment]
update: all the buffers that open by neo-tree will not be tracked, all the buffers that open by `:e file` works correctly.

**Comment by mrbeardad** [comment]
I have identified the issue, [utils/init.lua:667](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/main/lua/neo-tree/utils/init.lua#L667), `M.escape_path(path)` will replace all `\` with `/` on Windows, adding `path:gsub('/', '\\')` below it can solve the issue. Is there anybody can fix it elegantly? @nvim-neo-tree

**Comment by bwpge** [comment]
I just noticed I was having this same problem on a Windows system. Thanks for identifying the problem.

I was playing around with your suggested fix and found the `M.escape_path(path)` can be removed entirely, and we can directly use `vim.fn.fnameescape(path)`. I don't see any usage of it outside of the `utils/init.lua` -- quick `rg` for matches:

```
lua\neo-tree\utils\init.lua
667:    local escaped_path = M.escape_path(path)
1003:M.escape_path = function(path)
```

My only concern is that it seems other parts of the plugin depend on the buffer name, and I'm not sure where that string may be tossed around. On a Windows system, arguments and external programs that take a file path should handle the path separator (`\`) correctly, but I'm not familiar enough with the plugin to know what else might be affected.

Funny enough, this also fixes an issue I was having with `lualine.nvim` where the git branch wasn't being displayed when opening a file with Neo-tree. See https://github.com/nvim-lualine/lualine.nvim/issues/1183 for reference.

**Comment by bwpge** [comment]
After digging into my above comment a little more, this is a pretty big can of worms. Essentially boils down to (neo)vim not handling (escaped) Windows paths consistently. The trail starts in #889 and #1023.

I think I may open a separate issue and link it to others. Quickly searching open issues I could find at least 3 related to this, and found a bug with `set_root` (default keybind `.`) that hasn't been reported.

**Comment by bwpge** [comment]
@mrbeardad I believe this is fixed by #1353 which has been merged. If you want to double check you can pull the plugin spec from `main`.

**Comment by mrbeardad** [comment]
@bwpge Yeah, it works correctly on my Windows now, thanks!
